srtp-ekt-diet/draft-ietf-perc-srtp-ekt-diet.md

```diff
@@ -599,171 +599,201 @@ frames. If only sending audio, the RECOMMENDED frequency is every
+using EKT configured using the DTLS extensions described in this
+section.  (The initial cookie exchange is omitted.)
 
-{#tls-datastructure}
 ~~~
-enum {
-  reserved(0),  
-  aeskw_128(1),  
-  aeskw_256(2),  
-} EKTCipherType;  
+Client                                             Server
 
-struct {
-  EKTCipherType ekt_ciphers<1..255>;
-} SupportedEKTCiphers;
+ClientHello
+ + key_share*
+ + pre_shared_key*      -------->
+ + cookie
+ + use_srtp
+ + supported_ekt_ciphers
+                                               ServerHello
+                                              + key_share*
+                                         + pre_shared_key*
+                                                + use_srtp
+                                   + supported_ekt_ciphers
+                                     {EncryptedExtensions}
+                                     {CertificateRequest*}
+                                            {Certificate*}
+                                      {CertificateVerify*}
+                        <--------               {Finished}
 
-struct {
-  EKTCipherType ekt_cipher; 
-  uint ekt_key_value<1..256>;  
-  uint srtp_master_salt<1..256>;  
-  uint16 ekt_spi;  
-  uint24 ekt_ttl; 
-} EKTkey;  
 
+{Certificate*}
+{CertificateVerify*}
+{Finished}              -------->
+
+                        <--------                    [Ack]
+
+                        <--------                 [EKTKey]
+[Ack]                   -------->
+
+|SRTP packets|          <------->           |SRTP packets|
++ <EKT tags>                                  + <EKT tags>
+
+
+{} Messages protected using DTLS handshake keys
+
+[] Messages protected using DTLS application traffic keys
+
+<> Messages protected using the EKTKey and EKT cipher
+
+|| Messages protected using the SRTP Master Key sent in
+   a Full EKT Tag
+~~~
+
+When used in multi-party SRTP sessions, the DTLS Server may be a different
+entity than the source of the SRTP packets.  If the DTLS-SRTP server performs a
+handshake with each DTLS-SRTP endpoint to set a common EKT key, then each SRTP
+sender can use EKT to inform other participants of the keys it is using to
+send.  This avoids the need for many individual DTLS handshakes among the
+endpoints, at the cost of preventing endpoints from directly authenticating one
+another.
+
+~~~
+Client A                 Server                 Client B
+
+    <----DTLS Handshake---->
+    <--------EKTKey---------
+                            <----DTLS Handshake---->
+                            ---------EKTKey-------->
+
+    -------------SRTP Packet + EKT Tag------------->
+    <------------SRTP Packet + EKT Tag--------------
+~~~
+
+
+### Negotiating an EKT Cipher
+
+To indicate its support for EKT, a DTLS-SRTP client includes in its ClientHello
+an extension of type `supported_ekt_ciphers` listing the EKT ciphers the client
+supports.  If the server agrees to use EKT, then it includes a
```

Comment:
I think we need some text here about preferred cipher come first or something.

Reply:
In practice, I don't think this really matters, since servers do what they want.  But we can say it.